MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
+        boolean guidedAndArrowAvailable = ((year >= 2600) && (year <= 2835)) || (year > 3044);
+
+        // Generate a working map with all the unavailable ordnance replaced with rockets or HE
+        Map<Integer, Integer> workingBombMap = new HashMap<>();
+        for (int curBombType : bombMap.keySet()) {
+            String typeName = BombType.getBombInternalName(curBombType);
+            if (curBombType == BombType.B_RL ||
+                    curBombType == BombType.B_HE ||
+                    (curBombType != BombType.B_LG &&
+                            curBombType != BombType.B_ARROW &&
+                            curBombType != BombType.B_HOMING &&
+                            BombType.get(typeName).isAvailableIn(year, false)) ||
+                    ((curBombType == BombType.B_LG ||
+                            curBombType == BombType.B_ARROW ||
+                            curBombType == BombType.B_HOMING) &&
+                            guidedAndArrowAvailable)) {
+
+                if (workingBombMap.containsKey(curBombType)) {
+                    workingBombMap.put(curBombType, bombMap.get(curBombType) +
+                            workingBombMap.get(curBombType));
+                } else {
+                    workingBombMap.put(curBombType, bombMap.get(curBombType));
+                }
+
+            } else {
+
+                int replacementBomb = airOnly ? BombType.B_RL :
+                        Compute.randomInt(2) == 0 ? BombType.B_RL : BombType.B_HE;
+
+                if (workingBombMap.containsKey(replacementBomb)) {
+                    workingBombMap.put(replacementBomb, bombMap.get(curBombType) +
+                            workingBombMap.get(replacementBomb));
+                } else {
+                    workingBombMap.put(replacementBomb, bombMap.get(curBombType));
+                }
 
-            // hack: make rocket launcher pods available before 3055
-            if ((validBombChoices[x] == BombType.B_RL) ||
-                    BombType.get(typeName).isAvailableIn(year, false)) {
-                actualValidBombChoices.add(validBombChoices[x]);
             }
         }
 
-        // pick out the index in the BombType array
-        int randomBombChoiceIndex = Compute.randomInt(actualValidBombChoices.size());
-        int bombIndex = actualValidBombChoices.get(randomBombChoiceIndex);
-        int weightModifier = 0;
+        // Generate enough bombs to meet the desired count
+
+        int selectedBombType = -1;
+        int loopSafety = 0;
+        for (int curLoad = 0; curLoad < bombUnits && loopSafety < 10;) {
+
+            // Randomly get the ordnance type
+            completeWeight = workingBombMap.values().stream().mapToDouble(curWeight -> Math.max(curWeight, 1.0)).sum();
+            randomThreshold = (Compute.randomInt(100) / 100.0) * completeWeight;
+            countWeight = 0.0;
+            for (int curBomb : workingBombMap.keySet()) {
```

Comment:
See comments on similar block that selects specific maps from the map of roles and weights.